connection.go

```diff
@@ -248,18 +248,27 @@ func (c *conn) runQuery(ctx context.Context, query string, args []driver.NamedVa
 			cli_service.TOperationState_ERROR_STATE,
 			cli_service.TOperationState_TIMEDOUT_STATE:
 			logBadQueryState(log, statusResp)
-			return exStmtResp, statusResp, dbsqlerrint.NewDriverError(ctx, dbsqlerr.ErrInvalidOperationState, nil)
+			return exStmtResp, statusResp, unexpectedOperationState(statusResp)
 			// live states
 		default:
 			logBadQueryState(log, statusResp)
-			return exStmtResp, statusResp, dbsqlerrint.NewDriverError(ctx, dbsqlerr.ErrInvalidOperationState, nil)
+			return exStmtResp, statusResp, invalidOperationState(ctx, statusResp)
 		}
 	}
 }
 
 func logBadQueryState(log *logger.DBSQLLogger, opStatus *cli_service.TGetOperationStatusResp) {
 	log.Error().Msgf("databricks: query state: %s", opStatus.GetOperationState())
-	log.Error().Msg(opStatus.GetErrorMessage())
+	log.Error().Msg(opStatus.GetDisplayMessage())
+	log.Debug().Msg(opStatus.GetDiagnosticInfo())
+}
+
+func unexpectedOperationState(opStatus *cli_service.TGetOperationStatusResp) error {
+	return errors.WithMessage(errors.New(opStatus.GetDisplayMessage()), dbsqlerr.ErrUnexpectedOperationState(opStatus.GetOperationState().String()))
```

Comment:
Should we wrap this error in `dbsqlerrint.NewDriverError` or is that overkill?

Reply:
It isn't actually a driver error and it will be wrapped with an execution error back up the call chain so I don't think we should